crossterm_cursor/src/lib.rs

```diff
@@ -1,13 +1,449 @@
+//! ## Examples
+//!
+//! Basic usage:
+//!
+//! ```no_run
+//! // You can replace the following line with `use crossterm::TerminalCursor;`
+//! // if you're using the `crossterm` crate with the `cursor` feature enabled.
+//! use crossterm_cursor::{Result, TerminalCursor};
+//!
+//! fn main() -> Result<()> {
+//!     // Get a cursor, save position
+//!     let cursor = TerminalCursor::new();
+//!     cursor.save_position()?;
+//!
+//!     // Do something with the cursor
+//!     cursor.goto(10, 10)?;
+//!     cursor.blink(true)?;
+//!
+//!     // Be a good citizen, cleanup
+//!     cursor.blink(false)?;
+//!     cursor.restore_position()
+//! }
+//! ```
+//!
+//! Commands:
+//!
+//! ```no_run
+//! use std::io::{stdout, Write};
+//!
+//! use crossterm_cursor::{execute, Result};
+//! use crossterm_cursor::{BlinkOff, BlinkOn, Goto, ResetPos, SavePos};
+//!
+//! fn main() -> Result<()> {
+//!     execute!(
+//!         stdout(),
+//!         SavePos,
+//!         Goto(10, 10),
+//!         BlinkOn,
+//!         BlinkOff,
+//!         ResetPos
+//!     )
+//! }   
+//! ```
+use crossterm_utils::impl_display;
+#[cfg(windows)]
+use crossterm_utils::supports_ansi;
+#[doc(no_inline)]
 pub use crossterm_utils::{
     execute, queue, Command, ErrorKind, ExecutableCommand, Output, QueueableCommand, Result,
 };
-
-pub use self::cursor::{
-    cursor, BlinkOff, BlinkOn, Down, Goto, Hide, Left, ResetPos, Right, SavePos, Show,
-    TerminalCursor, Up,
-};
+use cursor::ansi::{self, AnsiCursor};
+#[cfg(windows)]
+use cursor::windows::WindowsCursor;
+use cursor::Cursor;
 
 mod cursor;
-pub mod sys;
+mod sys;
+
+/// A terminal cursor.
+///
+/// The `TerminalCursor` instance is stateless and does not hold any data.
+/// You can create as many instances as you want and they will always refer to the
+/// same terminal cursor.
+///
+/// The cursor position is 0 based. For example `0` means first column / row, `1`
+/// second column / row, etc.
+///
+/// # Examples
+///
+/// Basic usage:
+///
+/// ```no_run
+/// use crossterm_cursor::{Result, TerminalCursor};
+///
+/// fn main() -> Result<()> {
+///     let cursor = TerminalCursor::new();
+///     cursor.save_position()?;
+///
+///     cursor.goto(10, 10)?;
+///     cursor.blink(true)?;
+///
+///     cursor.blink(false)?;
+///     cursor.restore_position()
+/// }
+/// ```
+pub struct TerminalCursor {
+    #[cfg(windows)]
+    cursor: Box<(dyn Cursor + Sync + Send)>,
+    #[cfg(unix)]
+    cursor: AnsiCursor,
+}
+
+impl TerminalCursor {
+    /// Creates a new terminal cursor.
+    pub fn new() -> TerminalCursor {
+        #[cfg(windows)]
+        let cursor = if supports_ansi() {
+            Box::new(AnsiCursor::new()) as Box<(dyn Cursor + Sync + Send)>
+        } else {
+            Box::new(WindowsCursor::new()) as Box<(dyn Cursor + Sync + Send)>
+        };
+
+        #[cfg(unix)]
+        let cursor = AnsiCursor::new();
+
+        TerminalCursor { cursor }
+    }
+
+    /// Moves the cursor to the given position.
```

Comment:
take it or leave it

```suggestion
    /// Moves the cursor to the given position (column, row).
```

Reply:
Fixed by renaming arguments to `column` & `row`.

Reply:
actually that is way better than x, y